spec/use_cases/provider_key_change_use_case_spec.rb

```diff
@@ -85,20 +107,53 @@ module Backend
         end
 
         it 'sets the default service for the new provider key' do
+          use_case.process
           expect(Service.default_id(new_key))
               .to eq service_ids_with_old_key.first.to_s
         end
 
         it 'removes (old) provider key data' do
+          use_case.process
           expect(Service.default_id(old_key)).to be nil
           expect(Service.list(old_key)).to be_empty
         end
 
         it 'changes the provider key associated with the affected services' do
+          use_case.process
           service_ids_with_old_key.each do |service_id|
             expect(Service.provider_key_for(service_id)).to eq new_key
           end
         end
+
+        it 'clears the Service cache keys of the old provider key' do
+          service_ids_with_old_key.each do |service_id|
```

Comment:
this can be described in a new `context` or `describe` block if you split these examples in one part for the "general" specs and another one for the memoization stuff, and this way you would not need to insert `use_case.process` in all the other examples as a consequence of removing it from the `before` block (you would have two `describe`/`context` blocks, and each with independent `before` blocks if you do not nest them).